compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowExceptionHandlerFactory.desktop.kt

```diff
@@ -0,0 +1,37 @@
+package androidx.compose.ui.awt
+
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.ExperimentalComposeUiApi
+import java.awt.Window
+import java.awt.event.WindowEvent
+import javax.swing.JFrame
+import javax.swing.JOptionPane
+import javax.swing.SwingUtilities
+
+/**
+ * Default [WindowExceptionHandlerFactory], which will show standard error dialog, and close the window after that
+ */
+@ExperimentalComposeUiApi
+object DefaultWindowExceptionHandleFactory : WindowExceptionHandlerFactory {
+    override fun exceptionHandler(window: Window) = Thread.UncaughtExceptionHandler { _, throwable ->
+        SwingUtilities.invokeLater {
```

Comment:
What are thread safety expectations here, could it be invoked on non-main thread indeed? Shall we support such mode of operations?

Reply:
This function can be called only in UI thread. Added a comment about that